src/components/UserSelector.tsx

```diff
@@ -1,6 +1,62 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUsers } from '../api/users';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { UserOption } from './UserOption/UserOption';
+
+type Props = {
+  selectedId: number | null;
+  setSelectedUser: (user: User | null) => void;
+  loadPosts: (id: number) => void;
+  setIsLoading: (value: boolean) => void;
+  setError: (error: string) => void;
+  selectedUser: User | null;
+  setSelectedPost: (post: Post | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedId,
+  setSelectedUser,
+  loadPosts,
+  setIsLoading,
+  setError,
+  selectedUser,
+  setSelectedPost,
+}) => {
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [isSelectOpen, setIsSelectOpen] = useState(false);
+
+  useEffect(() => {
```

Comment:
Users should load when we load the app

Reply:
agree with the comment

Reply:
Also, when you open the list with options and click somewhere outside the options field, the dropdown must disappear